[APPC-5051] Wallet app log-in make minor changes for quick wins

arthur-oliveira-aptoide opened the pull request:
**What does this PR do?**
Improve the user experience by adding a connection badge logic to the Manage Wallets Fragment.
Allowing the user to easily know with wallet is associated with an email.
Also added the email information to the Home Fragment, displaying the email associated with the active wallet.

To support this implementation the `WalletInfoEntity` has been updated to also stores the email associated with 
that wallet.

**NOTE:** This PR also includes changes implemented in [APPC-5012](https://github.com/Catappult/appcoins-wallet-android/pull/1384)

**Database changed?**

   Yes

**Where should the reviewer start?**

- [ ] ManageWalletFragment.kt
- [ ] ChangeActiveWalletBottomSheetFragment.kt
- [ ] HomeFragment.kt
- [ ] CustomTabLoginActivity.kt
- [ ] WebViewLoginActivity.kt
- [ ] WebViewPaymentActivity.kt
- [ ] FetchUserKeyUseCase.kt
- [ ] UpdateWalletInfoUseCase.kt
- [ ] WalletInfoRepository.kt
- [ ] WalletInfoDao.kt

**How should this be manually tested?**

To make this test the Version code should be  >= **378**

Make a new login.

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-5051](https://aptoide.atlassian.net/browse/APPC-5051)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] Database migration (if applicable)?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass


[APPC-5051]: https://aptoide.atlassian.net/browse/APPC-5051?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[APPC-5012]: https://aptoide.atlassian.net/browse/APPC-5012?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

arthur-oliveira-aptoide committed:
[5051] Preview test

put all preview in comment

arthur-oliveira-aptoide committed:
[APPC-5051] Composable Function Refactor

Refactoring has been performed to enable proper Preview functionality.

arthur-oliveira-aptoide committed:
[APPC-5051] Connection Badge Implementation

The “ShowWalletsNameAndConnection” and “ConnectionBadge” functions have been implemented to enhance wallet connection validation capabilities.

arthur-oliveira-aptoide committed:
[APPC-5051] Wallet icon update.

arthur-oliveira-aptoide committed:
[APPC-5051] Enhance the Change Active Wallet Fragment by Incorporating a Wallet Connection Badge

**Changes Made:**

- A new argument has been incorporated into the fragment call.
- The `change_active_wallet_bottom_sheet_layout` XML has been updated to reflect the new UI.
- The `WalletInforSimple` has been updated to include the email value.
- Background and icons have been added to the project to support the new implementation.

arthur-oliveira-aptoide committed:
[APPC-5051] Wallet's email has been added to HomeView

arthur-oliveira-aptoide committed:
[APPC-5051] Switch account button has been implemented

When the user have a signed-in wallet in HomeFragment the sing-in button is replaced by a switch account.

arthur-oliveira-aptoide committed:
[APPC-5051] TODO: Check the other usages if UpdateWalletInfoUseCase

arthur-oliveira-aptoide committed:
[APPC-5051] code refactor

minor changes have been made to improve code's readability.

arthur-oliveira-aptoide committed:
Revert "[APPC-5051] code refactor"

This reverts commit 6671d45582bfbd991317ca08845d5d7e3d16c4e8.

arthur-oliveira-aptoide committed:
[APPC-5051] ManageWalletFragment update

arthur-oliveira-aptoide committed:
[APPC-5051] BalanceNewCard component update

arthur-oliveira-aptoide committed:
[APPC-5051] HomeFragment update

Logic to support email display has been implemented.

arthur-oliveira-aptoide committed:
[APPC-5051] WalletInfoEntity update.

email field has been added to `WalletInfoEntity`.

arthur-oliveira-aptoide committed:
[APPC-5051] WebViewPaymentInterface update.

email param has been add to OnLogin function

arthur-oliveira-aptoide committed:
[APPC-5051] Merge branch 'task/APPC-5012-Wallet-App-Log_In-Adjust-to-open-Custom-Chrome-Tab' into task/APPC-5051-Wallet-App-Log_in-Make-minor-changes-for-quick-wins

# Conflicts:
#	app/src/main/java/com/asfoundation/wallet/ui/webview_payment/WebViewPaymentInterface.kt

arthur-oliveira-aptoide committed:
[APPC-5051] Email addition to fetchUserKeyUseCase

arthur-oliveira-aptoide committed:
[APPC-5051] Code minor changes

Padding update.

arthur-oliveira-aptoide committed:
[APPC-5051] observeWalletsModel update

arthur-oliveira-aptoide committed:
[APPC-5051] Email addition to CCT and WebCheckout login flow.

arthur-oliveira-aptoide committed:
[APPC-5051] code minor changes

- code refactoring

rodolfobcardoso commented on feature/wallet-info/data/src/main/java/com/appcoins/wallet/feature/walletInfo/data/wallet/db/entity/WalletInfoEntity.kt:
```diff
@@ -23,4 +23,5 @@ data class WalletInfoEntity(
   val fiatCurrency: String?,
   val fiatSymbol: String?,
   val canTransfer: Boolean,
+  val email: String?
```
We must increase the DB scheme version since there is this new email field. It's needed to keep compatibility without needing to clear user data from the device.
We need to:
 - increase the scheme version
 - create a custom migration to treat the scenario where there is already a DB created and now the app tries to read this email field

arthur-oliveira-aptoide replied:
The migration 4_5 has been implemented, ensuring that old wallets can continue to work with the new DB.

rodolfobcardoso commented on app/src/main/java/com/asfoundation/wallet/ui/login/custom_tab_login/CustomTabLoginActivity.kt:
When performing a back press from the CCT, there is an activity that is kept on top, and blocks the user actions until he does another back press. Seems to be this activity.

rodolfobcardoso commented on feature/wallet-info/data/src/main/java/com/appcoins/wallet/feature/walletInfo/data/wallet/db/WalletInfoDao.kt:
```diff
@@ -101,6 +101,18 @@ interface WalletInfoDao {
+   * Attempts to insert a WalletInfoEntity with email. If it already exists, it just
+   * updates the email field.
+   */
+  fun insertOrUpdateEmail(walletInfo: WalletInfoEntity) {
```
When testing, it seems that the email is not updated when the wallet is already added.
To replicate:
 - use a wallet that don't have the login done, but is already stored in the email. (you can use the backup/recover wallet to store it).
 - with that wallet, try to login.
 - it is showing as if the login wasn't completed

rodolfobcardoso reviewed (changes requested):
Great code overall! Left some notes in the comments

arthur-oliveira-aptoide committed:
[APPC-5051] Database migration

arthur-oliveira-aptoide committed:
[APPC-5051] wallet update has been implemented.

When a wallet is already logged-in the email and name of the wallet is updated.

This is ensures that old wallets can also show new connection badge at new login.

arthur-oliveira-aptoide committed:
[APPC-5051] UI minor changes

- paddings values update.
- new color to currency text.

arthur-oliveira-aptoide committed:
[APPC-5051] Back from CCT fix.

When the user interrupts the login flow, going back to wallet,
the NativeLoginActivity is closed.

arthur-oliveira-aptoide committed:
[APPC-5051] FetchUserKeyUseCase update.

- setDefaultWalletUseCase has been removed.
- execution order has been changed.

arthur-oliveira-aptoide committed:
[APPC-5051] UI minor change.

Bottom spacer has been added to `ShowWalletsNameAndConnection`

arthur-oliveira-aptoide committed:
[APPC-5051] UI minor changes

- padding update on `BalanceBottomSheet`
- spacer update on `BalanceNewCard`.

arthur-oliveira-aptoide committed:
[APPC-5051] Switch account has been added to Settings fragment.

If the user have a logged in  wallet, the sing in text goes to switch account.
